### PR TITLE
LAND-1000: %genuine agent

### PR DESCRIPTION
**Pull request by midsum-salrux**
https://linear.app/tlon/issue/LAND-1000/s3-create-urbit-side-secret-for-tlon-hosting

New %genuine agent that creates a secret stored on your ship. Usage: scry secret from the frontend at `/secret` and pass it to hosting S3 or other services. Then that service can check `[ship.tlon.network]/genuine/[secret]` and get either json `true` or json `false` for a successful or failed match.

### CODE REVIEW

File: desk/app/genuine.hoon [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  |=  eny=@uvJ
+  ^-  cord
+  (scot %uw eny)
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-0
+  ==
++$  state-0
+  $:  %0
+      secret=cord
```
Comment (Fang-) [inline desk/app/genuine.hoon:16]: Arguably we should just store the atom here as-is, and only serialize on an as-needed basis. It'll be smaller to store and easier to reason about. ...Not my very much at all, for either of those, but storing your state in some serialized format feels less clean.

File: desk/app/genuine.hoon [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+%-  agent:dbug
+%+  verb  |
+^-  agent:gall
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %.n) bowl)
+::
+++  on-init
+  :_  this(secret (make-secret eny.bowl))
+  ~[[%pass /eyre/connect %arvo %e %connect [~ /genuine] dap.bowl]]
```
Comment (Fang-) [inline desk/app/genuine.hoon:31]: We might consider binding on `/apps/landscape/genuine` if we're not distributing this as a stand-alone application, just so we aren't squatting multiple top-level path element.

File: desk/app/genuine.hoon [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+++  on-watch
+  |=  =path
+  ^-  (quip card _this)
+  ?+  path  (on-watch:def path)
+    [%http-response *]  `this
+  ==
+::
+++  on-fail
+  |=  [=term =tang]
+  (mean ':sub +on-fail' term tang)
```
Comment (Fang-) [inline desk/app/genuine.hoon:63]:
```suggestion
  (mean ':genuine +on-fail' term tang)
```

Or `%genuine`, if you prefer.

File: desk/app/genuine.hoon [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    ~?  !accepted.sign-arvo
+      [dap.bowl 'eyre bind rejected!' binding.sign-arvo]
+    [~ this]
+  ==
+::
+++  on-peek
+  |=  =path
+  ^-  (unit (unit cage))
+  ?+  path  [~ ~]
+    [%x %secret ~]  ``json+!>([%s secret])
```
Comment (arthyn) [inline desk/app/genuine.hoon:88]: should we protect this? I know we share it outwardly, but maybe at the scry level it's worth protecting. not sure

**Review by arthyn — APPROVED** [review]
lgtm

**Review by Fang- — COMMENTED** [review]
I have my issues with the underlying reason/motivation/spec, but taking those as-is, this seems like it fulfills the requirements fine.

Some nits below. Also might be nice to have a quick paragraph at the top of the file describing what this is for/how it's supposed to be used.

File: desk/app/genuine.hoon (hunk 4 above)
Comment (Fang-) [reply desk/app/genuine.hoon:88]: `+on-peek` presently only gets called in cases where `=(our src):bowl`, so it's safe in that sense.

**Pull request merged**